medicines/web/tools/sitemap-generator.ts

```diff
@@ -31,27 +32,66 @@ const createPathsObj = async (): Promise<{ [index: string]: any }> => {
   );
 };
 
+const createSearchPathsObj = async (): Promise<{ [index: string]: any }> => {
+  // Get substance and product page URLs by mimicing the behavour of the A–Z, 0–9 list
+  // on the web site.
+  const searchPathsObj: { [index: string]: any } = {};
+
+  for (const letter of 'ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789') {
+    const facetResult = await facetSearch(letter);
+
+    for (const facet of facetResult[1].facets) {
+      // Value is in the format "letter, substance, product". Substance and product may
+      // not be present.
+      const stringParts = facet.value.split(', ');
+      const substance = stringParts[1];
+      const product = stringParts[2];
+
+      if (product) {
+        // If a product is present, output a product URL.
+        const route = `/?product=true&page=1&search=${encodeURIComponent(
```

Comment:
this routing logic is a duplication of where we do this in the application - i wonder whether there's a neat way we can share this route building